### DISCUSSION TITLE
Installation confusion

### DISCUSSION THREAD

**Discussion by gideunz** [CodyBerenson/PGMA-Modernized#262] [Q&A] [answered]
I am having a lot of trouble getting PGMA to work. I *think* the Installation and Usage guide leave some steps. Step 1 is "Clone this repository to your Plex server and copy all of the .bundle folders and their contents to your plugin folder," and Step 2 is "Exit and Restart your PMS and Login to the web interface and open settings," but there's no step to explain where all of the folders and files that don't end in .bundle go. Where do the folders titled ".github" "_PGMA" and "images" go? And where do the files titled ".git.ignore" "CODE_OF_CONDUCT.md" "CONTRIBUTING.md" "LICENSE" and "README.md" go? While it's clear that a bunch of these files have nothing to do with the actual operation of PGMA, it seems that just putting the bundles in the Plug-ins folder isn't enough -- because I can't get it to work. Scenes and films that are in the IAFD don't match with anything, no matter which agent I use. Anyway, any suggestions would be be most appreciated!

**Comment by CodyBerenson** [comment, marked as answer]
See Jason's answer in #226.   https://github.com/CodyBerenson/PGMA-Modernized/discussions/226#discussioncomment-6377382

Further, we wouldn't recommend you start with the IAFD agent to deem whether your setup is working or not.  IAFD has implemented cloud protections which renders the agent hit-or-miss, so you may not know whether your setup is correct without reviewing the log.  We would normally recommend starting with GEVI, but it too just implemented changes which broke the agent.  If you're interested in trying to match a film, try AEBN.  If you're interested in trying to match a scene, try WayBig.

**Reply by gideunz** [reply]
Thank you so much for that VERY quick answer. And thank you for this whole project! It seems that I've got it working now, and I can see how some of the sites are being difficult. The AEBN seems to be doing a good job getting the metadata, but the IAFD's actor thumbnails are a bit... wonky? 

I feel like the actor metadata has been an issue for everyone using Plex, but is there a way to manually fix stuff? Like if I wanted to use AEBN's thumbnails instead of IAFD's?

**Comment by JPH71** [comment]
I am working on GEVI this evening - so hopefully it will be working
tomorrow...

In the meantime - use AEBN for films and Waybig for Scenes as Cody has
suggested.

Cheers

Jason

**Comment by JPH71** [comment]
We stuck to using IAFDs actor thumbnails as AEBNs were not of standard size
and sometimes were whole body pics rather than face...
I am awaiting some code from a plex expert - so that users will be able to
store pictures under _PGMA that can be used instead of what is on IAFD...

IAFD seems to be tightening down with Cloudflare... sometimes it works like
a charm - at times it's like a pubescent angry teen....

We will hopefully find a way....

GEVI should be ready soon...


Jason

**Reply by fourstix** [reply]
The Adult nfo agent let's one point to a local source for images. I use a local instance of apache server to host images for nfo for movies not found on gevi or anywhere else.

**Reply by CodyBerenson** [reply]
@fourstix I had asked a different contributor how he creates his .NFOs.  https://github.com/CodyBerenson/PGMA-Modernized/issues/256#issuecomment-1597828026

Curious, do you use a tool to build your .NFOs or are you a notepad kinda guy?  I updated the usage guide with a little bit of discussion (beginning on page 28) relating to this agent and .NFOs, but always looking for more thoughts on how folks create their .NFOs.  Do you mind sharing your approach?  I've been tying to nudge Jason toward building 3rd party scraping templates for TinyMediaManager to move to .NFO only approach but he's a python addict and resistant. ;) LOL.

**Reply by fourstix** [reply]
@CodyBerenson  I'm a Notepad kinda guy.  It isn't too hard.  The format is simple and there's only a handful of fields that need to be filled out.  Mainly just the Title, Release Date, Plot, Summary, List of Genres, List of Actors.  I have the NFO agent configured to search my QNap NAS box for the actors.   I don't use a tool because most of these movies can't really be found anywhere by the other agents.

I have the NFO agent configured ping my QNap web server on the local network (inside my firewall) at http://192.168.1.230:8080/actors/ with the Actors Name with spaces replaced by an underscore.  

So for example Jon King will be http://192.168.1.230:8080/actors/Jon_King.jpg and I just create an actors directory in my NAS web directory with a jpg of Jon King's headshot named appropriately.

**Comment by dimmitri9** [comment]
I'm kinda confused :X the agent already on is the one that needs a fix due to today's site changes or is it ok?

**Reply by CodyBerenson** [reply]
Hi @dimitrimahiro I will try to provide clarity.

We now have two GEVI agents:  GEVI for films (one of the original agents in our set) and the brand-new GEVIScenes for scenes.  Simultaneous to Jason finishing tweaks to both and me uploading both to the master repository, GEVI overhauled their site's look and feel which broke both of our agents.  The fix to GEVIScenes was a quick one:  it has been coded, tested, and added to the master repository.  The fix to GEVI will require a MAJOR rewrite, so Jason believes the GEVI films agent may be out of commission for a while.  

In summary:  our GEVIScenes agent works, our GEVI agent does not.

Hope that helps.

**Reply by ScoBoXxX** [reply]
WTF, GEVI?!  Don't they know we're out here lurking in the background just trying to manipulate their service?? hehe
Now my 2 main sources aren't working - still can't use Fagalicious since I was deemed a scraper or whatever ha!
I'm building up quite the collection of files that hopefully will start matching soon :)
Thanks guys for all you do - has to be a labor of love (or lust) to be able to put up with all these changes.

**Reply by CodyBerenson** [reply]
LOL @ScoBoXxX.  I feel your pain.  Fagalicious has in a lot of ways bypassed Waybig as my favorite scenes site...  

Here's my consistent inconsistent experience with both Fagalicious and IAFD agents.  Both will rarely, but sometimes, match on the first attempt.  Both will eventually match after one...two...three...seven....fix match attempts.  But for me at least, both will still eventually match and scrape.....

Before writing this, I just tested half a dozen titles against Fagalicious.  Two actually matched and scraped on the initial scan that added the titles to the library!  Three matched and scraped on the first fix match.  One matched and scraped on the second fix match.  Quite consistently inconsistent...but at least for me, consistently (eventually) crosses the finish line.  And for me, it doesn't matter which (or no) VPN I am using at the moment.  Your experience may vary.

**Reply by ScoBoXxX** [reply]
My experience definitely varies lol.  Mine all fail hard, every time.  Log shows they fail with the much hated 403 Forbidden error from Fagalicious.  I'm still trying to get around that but no luck, VPN or not.  I'm probably gonna try and change my public IP with my provider next </shrug>.   They used to work great - 1st time almost every time.  Until they didn't anymore.
But yeah getting GEVI back would east that pain - ha :)

**Comment by JPH71** [comment]
GEVI is still being fixed at the moment.
When it is done I will send it together with all the recently updated Scene
Agents as the utils.py is kept at the same version for all the released
agents.
So once they are updated in the repository - you can download the following

   1. Best Exclusive Porn
   2. Fagalicious
   3. GEVI
   4. GEVIScenes
   5. QueerClick
   6. Fagalicious

As it stands, you can download and test any of the above 6 bar GEVI.....
I am just sorting out the xPath at the moment for the release dates and
duration....

Jason

**Comment by CodyBerenson** [comment]
Usage guide tweaked per @gideunz's helpful suggestion:

"Yeah, the Installation and Usage guide seems to leave out a bunch of steps in both the installtion and usage...it's clear that just putting the bundels in the Plug-ins folder isn't enough, because there is no magic happening at all. "

Update in bold:  Step 1:  Clone this repository to your Plex server and copy **the _PGMA folder and contents** and all of the .bundle folders and their contents to your plugin folder.

**Comment by JPH71** [comment]
My wee brown arse has been verbally smacked enough times... I will take a
look at this java thingy.... shudder...

With all this iafd... I think the time has come to be pulled out of my cavé
screaming...

**Comment by JPH71** [comment]
Hopefuly, I will get GEVI sorted out tonight....

I struggled last night and it scraped one movie perfectly - but no other
ones....

I must have set it up to scrape the one with the malformed html...

we will see....

GEVI is my main agent....

**Comment by JPH71** [comment]
@ScoBoXxX <https://github.com/ScoBoXxX> - GEVI sorted - with Cody for
testing!!

**Comment by JPH71** [comment]
In relation to GEVI, Cody should have posted it to the repository.
Please test and check when you get the new one... as GEVI has loads of
stuff and I have not managed to test the new format against many old
titles...

Many thanks

Jason xxx

**Comment by CodyBerenson** [comment]
Hi folks. Repository updated. see https://github.com/CodyBerenson/PGMA-Modernized/issues/261#issuecomment-1627740656

**Comment by ScoBoXxX** [comment]
I have a bunch of ‘vintage’ titles that i can check later today on the new GEVI agent.  I’ll download and update tonight sbsent from iThumbs; please pardon iTyposOn Jul 9, 2023, at 8:07 AM, Cody Berenson ***@***.***> wrote:﻿
Hi folks. Repository updated. see #261 (comment)

**Reply by JPH71** [reply]
Cheers man... much appreciated... xxx

**Reply by ScoBoXxX** [reply]
Confirmed.  I matched about 20 titles.  A few gave me a hard time but we sorted it out.  Some old ones are super finicky like you said.
Looking good - thanks!

sb

**Comment by JPH71** [comment]
Send me the list of the finicky twatty ones please....

**Reply by ScoBoXxX** [reply]
Sure thing!  Here they are.  I am able to search and find them on GEVI but they didn't match on the initial load or on a Fix Match attempt.

(Blade Productions) - What The Wife Doesn't Know (1999)
(P.M. Productions) - Young Yankees (1984)
(Venus Video) - Unchained Men (1989)
(Video 10) - Warlords (1989)
